### PR TITLE
[VS Incentives]: queries

### PR DESCRIPTION
**Pull request by czarcas7ic**
Partially Closes: #6154 

## What is the purpose of the change

This PR implements all queries described in issue 6154.

The only thing that initially did not behave as expected was the total volume query only returned in terms of uosmo, but thinking about it further I am pretty sure that is expected.

## Testing and Verifying

localosmosis

```
osmosis % osmosisd q incentives all-groups
groups:
- group_gauge_id: "10"
  internal_gauge_info:
    gauge_records:
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "3"
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "6"
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "9"
    total_weight: "0"
  splitting_policy: ByVolume
```
```
osmosis % osmosisd q incentives all-groups-gauges
gauges:
- coins:
  - amount: "10000000"
    denom: uosmo
  distribute_to:
    denom: ""
    duration: 0s
    lock_query_type: ByGroup
    timestamp: "0001-01-01T00:00:00Z"
  distributed_coins: []
  filled_epochs: "0"
  id: "10"
  is_perpetual: true
  num_epochs_paid_over: "0"
  start_time: "2023-09-29T14:21:35.870386501Z"
```
```
osmosis % osmosisd q incentives all-groups-with-gauge
groups_with_gauge:
- gauge:
    coins:
    - amount: "10000000"
      denom: uosmo
    distribute_to:
      denom: ""
      duration: 0s
      lock_query_type: ByGroup
      timestamp: "0001-01-01T00:00:00Z"
    distributed_coins: []
    filled_epochs: "0"
    id: "10"
    is_perpetual: true
    num_epochs_paid_over: "0"
    start_time: "2023-09-29T15:09:46.693365256Z"
  group:
    group_gauge_id: "10"
    internal_gauge_info:
      gauge_records:
      - cumulative_weight: "0"
        current_weight: "0"
        gauge_id: "3"
      - cumulative_weight: "0"
        current_weight: "0"
        gauge_id: "6"
      - cumulative_weight: "0"
        current_weight: "0"
        gauge_id: "9"
      total_weight: "0"
    splitting_policy: ByVolume
```

```
osmosis % osmosisd q incentives group-by-group-gauge-id 10
group:
  group_gauge_id: "10"
  internal_gauge_info:
    gauge_records:
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "3"
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "6"
    - cumulative_weight: "0"
      current_weight: "0"
      gauge_id: "9"
    total_weight: "0"
  splitting_policy: ByVolume
```

```
osmosis % osmosisd q poolmanager total-volume-for-pool 1
volume:
- amount: "39999998"
  denom: uosmo
```

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/incentives/keeper/grpc_query.go [hunk 1]
```diff
@@ -176,6 +176,49 @@ func (q Querier) LockableDurations(ctx context.Context, _ *types.QueryLockableDu
...
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	groupsWithGauge, err := q.Keeper.GetAllGroupsWithGauge(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryAllGroupsWithGaugeResponse{GroupsWithGauge: groupsWithGauge}, nil
+}
+
+func (q Querier) GroupByGroupGaugeID(goCtx context.Context, req *types.QueryGroupByGroupGaugeIDRequest) (*types.QueryGroupByGroupGaugeIDResponse, error) {
```
Comment (p0mvn) [inline x/incentives/keeper/grpc_query.go:212]: nit: godoc

File: x/incentives/keeper/store.go [hunk 2]
```diff
@@ -143,6 +143,33 @@ func (k Keeper) GetAllGroupsGauges(ctx sdk.Context) ([]types.Gauge, error) {
 	return gauges, nil
 }
 
+// GetAllGroupsWithGauge iterates through all groups, sequentially pulls the gauges from each, and returns the groups with their associated gauge.
+func (k Keeper) GetAllGroupsWithGauge(ctx sdk.Context) ([]types.GroupsWithGauge, error) {
```
Comment (p0mvn) [inline x/incentives/keeper/store.go:147]: Nice

**Review by p0mvn — APPROVED** [review]
Awesome. Nice work on these!

File: x/incentives/keeper/store.go (hunk 2 above)
Comment (p0mvn) [inline x/incentives/keeper/store.go:147]: Let's also cover this with a simple test plase

File: x/incentives/keeper/grpc_query.go (hunk 1 above)
Comment (czarcas7ic) [reply x/incentives/keeper/grpc_query.go:214]: https://github.com/osmosis-labs/osmosis/pull/6594/commits/42eee6903ed44dd714b75da22966e30d3da355c6

File: x/incentives/keeper/store.go (hunk 2 above)
Comment (czarcas7ic) [reply x/incentives/keeper/store.go:147]: done https://github.com/osmosis-labs/osmosis/pull/6594/commits/aa8c9e4bff03161ef00e495e1254106a0d77e54b

**Pull request merged**